app/services/download_service.rb

```diff
@@ -23,9 +23,10 @@ def download_windows_versions
         next unless match
 
         portable = match[1]
+        version_name = match [2]
         bitness  = match[3]
 
-        version = find_or_create_version_by_name(name)
+        version = find_or_create_version_by_name(version_name)
```

Comment:
Looking at 0f4974ec4cfbf3c5c897b2c7e2357d23f1002fa0, this fix is definitely the right direction. But I think there are a few things left to tweak.

The [original code](https://github.com/git/git-scm.com/commit/0f4974ec4cfbf3c5c897b2c7e2357d23f1002fa0#diff-c9c6fbfb07fabf866b3c6ad495128f37L61) there did a further `slice` on the version to cut off anything after the third level of numbers, though. I think we'd need that for cases like `2.16.1.4`, which should be filed under `2.16.1` (the GfW project sometimes creates extra releases all based off of the same upstream Git version, so we'd just want to make sure we're linking to the most recent one that's `v2.16.1.x` in that case).

The original also didn't do a `find_or_create`, but instead relied on the `preindex` task to have created an entry for the version already. In an ideal world _either_ could create the entry, so we didn't have any kind of dependency. Worst case with the old code is that the `downloads` task might do nothing until `preindex` runs, delaying the update of the link for a day. But with this `find_or_create`, I think we can get into a funny situation where the `download` task creates the version record, and then later `rake preindex` thinks that we've already imported the docs and does nothing (due to [this code](https://github.com/git/git-scm.com/blob/6dee71fb1eb917de1beb6935a9bb5ed8cadc2c58/lib/tasks/index.rake#L15-L16)). I think we'd need to make that check more careful (e.g., by adding a new column that says "yes, we've imported the docs for that version"). Or possibly just separate the downloads and manpage versions altogether. I don't think they ever need to interact in a meaningful way.  Perhaps the original downloads code was just being careful that it did not import a false positive that did not even have a matching Git tag.

Reply:
It should be easy to solve the extra release problem.

About the conflict with the index task, I will need to have a better look. I'll try to finish this during this week

thanks for the feedback